src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -27,6 +28,8 @@ internal class StartupHook
 {
     private static readonly IOtelLogger Logger = OtelLogging.GetLogger("StartupHook");
 
+    internal static string? LoaderAssemblyLocation { get; set; }
```

Comment:
Add a comment that this property must be initialized before any rule is evaluated since they may use it. In light of that it seems better to initialize its value here.